lib/prom_ex/config.ex

```diff
@@ -129,6 +129,9 @@ defmodule PromEx.Config do
         - Git SHA of the last commit (if the GIT_SHA environment variable is present)
         - Git author of the last commit (if the GIT_AUTHOR environment variable is present)
 
+    * `:finch_pools` - (optional) A map that will be passed to Finch.start_link/1 as the :pools key,
```

Comment:
I opted for `finch_pools` rather than `finch: [ pools: ... ]` for two reasons:

1. [`Finch.start_link/1`](https://hexdocs.pm/finch/Finch.html#start_link/1) only defines two options: `name` and `pools`. If they add more options at some point, this could get messy - but if that happened, we could consider changing this to just `finch`.
2. More importantly, [`PromEx.grafana_client_child_spec/4`](https://github.com/akoutmos/prom_ex/blob/d51b974d2ebcacdef1d9044693a3b7c5211c2c6b/lib/prom_ex.ex#L416) passes `:name` to [`GrafanaClient.child_spec/`](https://github.com/akoutmos/prom_ex/blob/d51b974d2ebcacdef1d9044693a3b7c5211c2c6b/lib/prom_ex/grafana_client.ex#L31), which passes it to Finch.

This would lead to some ambiguity if you did e.g. `finch: [ name: MyApp, pools: %{ ... }]`.

It's possible to address - maybe just set a default name if the Finch options key doesn't exist. I just don't know that it's worth it at this point.

Reply:
I think the way that you have it now it fine given that finch doesn't provide any other configuration options that need to be leveraged. Will merge in as soon as CI passes :)